services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/student_report.tsx

```diff
@@ -26,119 +28,183 @@ interface StudentReportProps extends RouteComponentProps {
     studentId: string,
     unitId: string
   },
-  studentDropdownCallback: () => void
+  studentDropdownCallback: () => void,
+  passedStudents?: Student[]
 }
 
-export class StudentReport extends React.Component<StudentReportProps, StudentReportState> {
-
-  state = {
-    boldingExplanationIsOpen: false,
-    loading: true,
-    students: null
-  };
-
-  componentDidMount() {
-    const { params } = this.props;
-    this.getStudentData(params);
-  }
-
-  componentDidUpdate(prevProps) {
-    const { params } = this.props;
-
-    if (!_.isEqual(params, prevProps.params)) {
-      this.setState({ loading: true });
-      this.getStudentData(params);
+const StudentReport = ({ params, studentDropdownCallback, passedStudents, }) => {
+  const [boldingExplanationIsOpen, setBoldingExplanationIsOpen] = React.useState(false)
+  const [scoringExplanationIsOpen, setScoringExplanationIsOpen] = React.useState(false)
+  const [loading, setLoading] = React.useState(!passedStudents)
+  const [students, setStudents] = React.useState(passedStudents || null)
+
+  const isInitialMount = React.useRef(true);
+
+  React.useEffect(() => {
+    if (isInitialMount.current) {
+      isInitialMount.current = false
+      getStudentData()
+    } else {
+      setLoading(true)
+      getStudentData()
     }
-  }
+  }, [params])
```

Comment:
just curious why you opted to go with this `isInitialMount` variable reference rather than using `[]` as an arg in `useEffect`?

Reply:
Using `[]` would have it fire only on initial load, but we want to make sure this gets called both on initial load and also when the params change. However, we only want it to `setLoading(true)` on later loads, so we need that bit of logic in there to see if it's the first time it's loaded or not.